### ISSUE TITLE
Weird redirection with PHP > 7.0

### ISSUE THREAD

**Issue by elbaku** [benedmunds/CodeIgniter-Ion-Auth#1069]
Hello @benedmunds, i'm updating an app with Ion Auth and CI 3.1.0 on a DevEnv with PHP 5.6.30, i just changed the PHP version to 7.1.1 and the login looks like its not saving the session. I downloaded again the IonAuth library and installed it again from scratch (files and db), when the code executes the following line:

```
$this->ion_auth->login($user, $password, $remember);
```

it returns true with the default user, but when i check again the session with:

```
if (!$this->ion_auth->logged_in())
```

it returns false and redirects me to the login page again, this does not happen when i activate de "Remember me" checkbox, the weirdest thing is that if i change the PHP Version to 5.6.30 again it works flawlessly, can you help me? What info of my DevEnv do you need?

Thank you!
F.

**Comment by benedmunds** [comment]
Can you post the following files, or links to gists of them:

config/config.php
config/ion_auth.php

Make sure to obfuscate any sensitive information.



Also can you post the output of:
$ php -i

Make sure the php being used is the PHP 7 instance.

**Comment by avenirer** [comment]
Can you first try to use CI 3.1.4?

**Comment by elbaku** [comment]
Thank you @avenirer it looks like updating the CI version was the right call, login now works with/without the "remember me" option, do you know why this happened? Session updates on de CI?

Thank you both @benedmunds and @avenirer for your time.
F.

**Comment by avenirer** [comment]
There was a fix regarding regenerate session in 3.1.4

**Comment by elbaku** [comment]
Thank you @avenirer i will close the ticket
